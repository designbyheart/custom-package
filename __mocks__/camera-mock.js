import React from 'react'
import { View } from 'react-native'

const constants = (constants = {
  Aspect: {},
  BarCodeType: {},
  Type: {},
  CaptureMode: {},
  CaptureTarget: {},
  CaptureQuality: {},
  Orientation: {},
  FlashMode: {},
  TorchMode: {},
})

class MockCamera extends React.Component {
  static constants = constants

  render() {
    return <View>{this.props.children}</View>
  }
}

MockCamera.constants = constants

export default MockCamera
