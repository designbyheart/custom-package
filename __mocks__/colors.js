export const colors = {
  cmGreen1: '#86B93B',
  cmGreen2: '#6C8E3A',
  cmGreen3: 'rgba(134, 185, 59, 0.15)',
  cmRed: '#CE0B24',
  cmOrange: '#EB9B2D',
  cmWhite: '#FFFFFF',
  cmGray5: '#F2F2F2',
  cmGray4: '#EAEAEA',
  cmGray3: '#A5A5A5',
  cmGray2: '#777777',
  cmGray1: '#505050',
  cmBlack: '#000000',
  cmBlue: '#236BAE',
}
