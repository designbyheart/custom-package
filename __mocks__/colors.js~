export const colors = {
  cmGreen1: '#0000ff',
  cmGreen2: '#0000ff',
  cmGreen3: 'rgba(0, 0, 255, 0.15)',
  cmRed: '#ff0000',
  cmOrange: '#EB9B2D',
  cmWhite: '#FFFFFF',
  cmGray5: '#F2F2F2',
  cmGray4: '#EAEAEA',
  cmGray3: '#A5A5A5',
  cmGray2: '#777777',
  cmGray1: '#505050',
  cmBlack: '#000000',
  cmBlue: '#236BAE',
}

