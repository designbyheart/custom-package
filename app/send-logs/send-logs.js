// @flow

import React, { PureComponent } from 'react'
import {
  StyleSheet,
  Image,
  NativeModules,
  View,
  ScrollView,
  Text,
  Button,
  Platform,
  TextInput,
  Alert,
  Dimensions,
} from 'react-native'
import type { SendLogsProps } from './type-send-logs'
import type { Store } from '../store/type-store'
import { connect } from 'react-redux'
import {
  Container,
  CustomView,
  CustomHeader,
  Icon,
  CustomText,
} from '../components'
import {
  color,
  OFFSET_1X,
  OFFSET_2X,
  OFFSET_6X,
  OFFSET_7X,
  dimGray,
  lightGray,
  grey,
} from '../common/styles'
import { bindActionCreators } from 'redux'
import { selectUserAvatar } from '../store/user/user-store'
import { createStackNavigator, NavigationActions } from 'react-navigation'
import { homeRoute, sendLogsRoute } from '../common/route-constants'
import Mailer from 'react-native-mail'
import { customLogger } from '../store/custom-logger'
import { Loader } from '../components'
import { getLogEncryptionStatus } from '../store/store-selector'
import {
  UPDATE_LOG_ISENCRYPTED,
  ENCRYPT_LOG_FILE,
} from '../send-logs/type-send-logs'
import store from '../store'

const width = Dimensions.get('window').width //full width
const height = Dimensions.get('window').height //full height

const styles = StyleSheet.create({
  headerLeft: {
    width: OFFSET_2X,
  },
  headerRight: {
    width: OFFSET_2X,
  },
  inputBox: {
    backgroundColor: lightGray,
    padding: 10,
    textAlignVertical: 'top',
    fontSize: 18,
    borderColor: grey,
    borderLeftWidth: 1,
    borderRightWidth: 1,
    borderTopWidth: 1,
    borderBottomWidth: 1,
    height: height * 0.4,
    width: width * 0.8,
  },
})

export class SendLogs extends PureComponent<SendLogsProps, void> {
  constructor(props: SendLogsProps) {
    super(props)
  }

  componentDidMount() {
    store.dispatch({
      type: ENCRYPT_LOG_FILE,
    })
  }

  //static emailMessageBody = ''

  static handleEmail = () => {
    const logFile = customLogger.getEncryptedVcxLogFile()
    Mailer.mail(
      {
        subject: 'ConnectMe Application Log: ' + logFile,
        recipients: ['cmsupport@evernym.com'],
        body: '',
        isHTML: false,
        attachment: {
          path: logFile, // The absolute path of the file from which to read data.
          type: 'text', // Mime Type: jpg, png, doc, ppt, html, pdf, csv, vcard, json, zip, text, mp3, wav, aiff, flac, ogg, xls
          name: logFile, // Optional: Custom filename for attachment
        },
      },
      (error, event) => {
        const sentButton = {
          text: 'OK',
          onPress: () => {
            //customLogger.log('SENT: ConnectMe Logs via Email')
          },
        }
        const cancelButton = {
          text: 'OK',
          onPress: () => {
            //customLogger.log('CANCELLED: ConnectMe Logs NOT sent via Email')
          },
        }
        const notAvailableMsg =
          'Unable to send error report. Sending error logs requires you to be signed into at least one email account on the default mail app which came with the phone out of the factory (the "native" mail app). Please sign into an email account from the default email app that came with the phone out of the factory and try again.'
        const cancelSendMsg = 'You did not send error logs to Evernym.'
        let alertMsg = event
        let alertButton = cancelButton
        if ('not_available' === error) {
          alertMsg = notAvailableMsg
        } else if ('cancelled' === event) {
          alertMsg = cancelSendMsg
        } else if ('sent' === event) {
          alertMsg = 'Sent'
          alertButton = sentButton
        }

        Alert.alert('', alertMsg, [alertButton], { cancelable: true })
      }
    )
  }

  static goBack(navigation: any) {
    //Alert.alert('NOT sending logs')
    navigation.goBack(null)
    setTimeout(() => {
      store.dispatch({
        type: UPDATE_LOG_ISENCRYPTED,
        logIsEncrypted: false,
      })
    }, 3000)
  }

  static sendLogs(navigation: any) {
    //Alert.alert('sending logs from file: ' + String(customLogger.getVcxLogFile()))
    SendLogs.handleEmail()
    SendLogs.goBack(navigation)
  }

  static navigationOptions = ({ navigation }) => ({
    header: (
      <CustomHeader
        backgroundColor={color.bg.tertiary.color}
        outerContainerStyles={{ borderBottomWidth: 0 }}
      >
        <Icon
          testID={'send-logs-close-icon'}
          iconStyle={[styles.headerLeft]}
          src={require('../images/icon_close.png')}
          resizeMode="contain"
          onPress={() => SendLogs.goBack(navigation)}
          small
        />
        <Icon
          testID={'send-logs-send-icon'}
          iconStyle={[styles.headerRight]}
          src={require('../images/iconRArrow.png')}
          resizeMode="contain"
          onPress={() => SendLogs.sendLogs(navigation)}
          small
        />
        <CustomText bg="tertiary" tertiary transparentBg semiBold />
        <CustomView />
      </CustomHeader>
    ),
    swipeEnabled: false,
  })

  render() {
    return this.props.logIsEncrypted ? (
      <Container tertiary>
        <CustomView center>
          <CustomText bg="secondary" secondary transparentBg semiBold>
            Send error logs to cmsupport@evernym.com ?
          </CustomText>
        </CustomView>
      </Container>
    ) : (
      // Show spinner until the log file is encrypted...
      <Container center>
        <Loader message="Encrypting log..." />
      </Container>
    )
  }
}

const mapStateToProps = (state: Store) => {
  return {
    logIsEncrypted: getLogEncryptionStatus(state),
  }
}

export const SendLogsStack = createStackNavigator({
  [sendLogsRoute]: {
    screen: connect(mapStateToProps)(SendLogs),
  },
})

export default SendLogsStack
