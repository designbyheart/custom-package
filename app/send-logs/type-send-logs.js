// @flow
import type { ReactNavigation } from '../common/type-common'

export type SendLogsProps = {} & ReactNavigation
