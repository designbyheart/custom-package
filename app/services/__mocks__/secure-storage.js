let secureStorage = {
  ConnectMeKeyChain: {},
}

export const setItem = (key, data) => {
  secureStorage['ConnectMeKeyChain'][key] = data
}

export const getItem = key =>
  Promise.resolve(secureStorage['ConnectMeKeyChain'][key])

export const deleteItem = key => delete secureStorage['ConnectMeKeyChain'][key]
