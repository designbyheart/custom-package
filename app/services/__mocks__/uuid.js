// @flow

export function uuid() {
  return 'a4f35623-b50c-40ea-a2b0-f7cd06e03142'
}
