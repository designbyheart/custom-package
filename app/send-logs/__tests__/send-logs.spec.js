// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { SendLogs } from '../send-logs'
import { getStore } from '../../../__mocks__/static-data'
import { getNavigation } from '../../../__mocks__/static-data'

import type {} from '../type-send-logs'

describe('send logs screen', () => {
  const props = {
    navigation: getNavigation(),
    environmentName: 'DEMO',
    logIsEncrypted: false,
    route: {},
  }

  it('should render properly and snapshot should match', () => {
    const tree = renderer.create(<SendLogs {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
