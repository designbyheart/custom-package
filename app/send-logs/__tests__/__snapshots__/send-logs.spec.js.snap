// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`send logs screen should render properly and snapshot should match 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    center={true}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          undefined,
          Object {
            "fontWeight": "600",
          },
          null,
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
        ]
      }
    >
      Send error logs to cmsupport@evernym.com ?
    </Text>
  </View>
</View>
`;
